x-pack/functionbeat/function/telemetry/telemetry.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package telemetry
+
+import (
+	"strconv"
+	"sync"
+
+	"github.com/elastic/beats/libbeat/monitoring"
+)
+
+// T is a telemetry instance
+type T interface {
+	AddTriggeredFunction(string, int64, int64)
+}
+
+type telemetry struct {
+	sync.Mutex
+	nextID int
+
+	registry *monitoring.Registry
+}
+
+// New returns a new telemetry registry.
+func New(r *monitoring.Registry) T {
+	return &telemetry{
+		nextID:   0,
+		registry: r.NewRegistry("functions"),
+	}
+}
+
+// AddTriggeredFunction adds a triggered function data to the registry.
+func (t *telemetry) AddTriggeredFunction(typeName string, triggerCount, eventCount int64) {
```

Comment:
What is the purpose of triggerCount or eventCount? Telemetry is less about monitoring, we just want to know which features are used.

Reply:
I was interested to see how many triggers users configure. I can remove both counters if we don't want those to be reported at all.